core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -206,6 +210,11 @@ private[spark] object Task {
       dataOut.writeLong(timestamp)
     }
 
+    // Write the task properties separately so it is available before full task deserialization.
```

Comment:
Since the properties aren't transient in `Task`, I guess this means that we'll write them out twice. If we want to avoid this, we can make `localProperties` into a `@transient` `var` which is `private[spark]` then re-set the field after deserializing the task. Tasks are send to executors using broadcast variables, so the extra space only makes a different for the first task from a stage that's run on an executor.

As a result, if we think that these serialized properties will typically be small then the extra space savings probably aren't a huge deal, but if we want to heavily optimize then we can do the `var` trick.